extends CharacterBody3D

const SPEED = 5.0
const JUMP_VELOCITY = 4.5
var banana_counter = 0

@onready var banana_label = %Label

func _physics_process(delta: float) -> void:
	# Add the gravity.
	if not is_on_floor():
		velocity += get_gravity() * delta

	# Handle jump.
	if Input.is_action_just_pressed("jump") and is_on_floor():
		velocity.y = JUMP_VELOCITY

	# Get the input direction and handle the movement/deceleration.
	# As good practice, you should replace UI actions with custom gameplay actions.
	var input_dir := Input.get_vector("left", "right", "forward", "back")
	var direction := (transform.basis * Vector3(input_dir.x, 0, input_dir.y)).normalized()
	if direction:
		velocity.x = direction.x * SPEED
		velocity.z = direction.z * SPEED
	else:
		velocity.x = move_toward(velocity.x, 0, SPEED)
		velocity.z = move_toward(velocity.z, 0, SPEED)

	move_and_slide()
var score:= 0
func sumar_score(cantidad):
	score+= cantidad
	print("score", score)
func _on_area_3d_area_entered(area: Area3D) -> void:
	if area.is_in_group("banana"):
		set_banana(banana_counter + 1)
		print(banana_counter)
		
func set_banana(new_banana_count: int) -> void:
	banana_counter = new_banana_count
	banana_label.text = "Banana Count: " + str(banana_counter)
	
